### ISSUE TITLE
Not able to retrieve thread data

### ISSUE THREAD

**Issue by gaganmeet** [KartikTalwar/gmail.js#684]
I was trying to inject this code 
`setInterval(() => {
  const email = gmail.new.get.email_id()
  const threadId = gmail.new.get.thread_id()
  const emailData = gmail.new.get.thread_data(threadId)
  console.log(email, emailData, threadId)
}, 5000)`
i made sure to inject this before document is loaded but still getting null for thread data. Also i am able to retrieve the threadId and emailId.

**Comment by josteink** [comment]
Does this apply to all threads or just the initially visited ones?

Some thread and email data is seeded early in the HTML, and you need to inject GmailJS early to pick those up.

**Comment by gaganmeet** [comment]
Yeah tried with different emails. Also tried injecting GmailJS before document loads still the same result.

**Comment by josteink** [comment]
I can't make any promises here, but if you can provide a minimal proof of concept extension/repo which I can clone, build and test and also include instructions for what to do to consistently reproduce this behaviour...

Then I can either see if there are errors in the way gmailjs is used, or if it is indeed a bug/lacking in the library.

Without that, I won't be able to help out much I'm afraid.

**Comment by josteink** [comment]
There has been recent bugfixes for this kind of issue in the latest published version (v1.0.21).

Can you test that and see if that fixes your issue? :)
